cmd: add --dedup-stats flag to ls; show per-file dedup stats

Janmesh23 opened the pull request:
This PR adds a `--dedup-stats` (`-d`) flag to `sietch ls`. When enabled, `ls` will:

- Show number of shared chunks per file (`shared_chunks`).
- Show estimated bytes avoided per file due to dedup (`saved`) using per-chunk EncryptedSize (fallbacks to Size, then 4MiB default).
- List other files that share those chunks (`shared_with`) — truncated to first 10 entries.

Notes:
- The saved bytes reflect how much duplicate data the file avoided storing (per-file view).
- For global reclaimed space metrics we could show `(refs-1)*chunkSize` aggregated per chunk; can be implemented in a follow-up PR.
- Includes unit test for `computeDedupStatsForFile`.

<img width="596" height="298" alt="Screenshot 2025-10-05 at 9 03 25 PM" src="https://github.com/user-attachments/assets/100c5498-654f-4b41-9b97-1f448fd91364" />

- The above image shows the implementation of the cmd   `sietch ls --dedup-stats`

<img width="888" height="254" alt="image" src="https://github.com/user-attachments/assets/1ef59e43-7b4f-42e2-b27b-a4a699ad6b48" />

- The above image shows the implementation of the cmd   `sietch ls -l -d`

Fixes: #96

SubstantialCattle5 commented:
![Uploading Screenshot_20251005-213646.png …]()
It's within this pr itself.

SubstantialCattle5 commented:
The Codecov report shows that the patch coverage for this PR is only 2.94%, with 99 lines in `cmd/ls.go` missing test coverage.
This means that most of the new or modified code in this PR isn’t covered by automated tests.

Janmesh23 commented:
> The Codecov report shows that the patch coverage for this PR is only 2.94%, with 99 lines in `cmd/ls.go` missing test coverage. This means that most of the new or modified code in this PR isn’t covered by automated tests.

sure , i'll get on it

SubstantialCattle5 commented:
> > <img alt="image" width="1705" height="997" src="https://private-user-images.githubusercontent.com/92233871/497566073-cefa0942-1ad9-427e-a328-2b2618b620c4.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTk3Mjc3NzMsIm5iZiI6MTc1OTcyNzQ3MywicGF0aCI6Ii85MjIzMzg3MS80OTc1NjYwNzMtY2VmYTA5NDItMWFkOS00MjdlLWEzMjgtMmIyNjE4YjYyMGM0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDA2VDA1MTExM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE0NzFiM2I3MTBiZjM3ZDI2OTkxMTY1Y2JkOTE0YTBlYmExODAwNTdjOTA2MmFiMDRiYWNmYzk1MTQ1YzYyMjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.hxf0bC0-cRDMVh1-ScdmWc9UUE-1QA4TYcVe81aIvKg">
> > there should be 3-4 comments tagged with review.
> 
> hi @SubstantialCattle5 , I could not see the comments you added in file changes please let me know where I can see them sorry for the inconvenience

@Janmesh23 , 
hi sure no worries I'll put them here itself. 

<img width="1382" height="541" alt="image" src="https://github.com/user-attachments/assets/27343bbb-2133-4ee0-8f13-ce7b11c3c48f" />

<img width="1382" height="541" alt="image" src="https://github.com/user-attachments/assets/3317b0d9-e7d4-4b62-aaa2-6aad29083fb7" />

<img width="1382" height="541" alt="image" src="https://github.com/user-attachments/assets/e3a5b54d-57db-4ade-b91c-1ce6fdf55dd3" />

Janmesh23 commented:
Refactored dedup logic for better modularity and testability.
Exported `ComputeDedupStatsForFile`, `FormatSharedWith`, and `DisplayShortFormat` so cmd and tests can reuse them.
Moved UI helpers under ` internal/ls` and kept dedup logic in ` internal/deduplication`.

SubstantialCattle5 commented:
@Janmesh23,

you’ve committed your test vaults and `bak` files as well. Please remove them.

Cheers,
Nilay

Janmesh23 commented:
> @Janmesh23,
> 
> you’ve committed your test vaults and `bak` files as well. Please remove them.
> 
> Cheers, Nilay

yeah my bad , doing it right away  :)

Janmesh23 commented:
> @Janmesh23, I've squashed your commits from my side and also updated the commit message to follow the guidelines. Please make sure to follow the standards defined for subsequent PRs.
> 
> Cheers, Nilay

yess sure !
thankyou for the opportunity @SubstantialCattle5 , i would like to work on more such issues in sietch and learn more .
looking forward to do so :)
